### PR TITLE
Improve device RFGen

### PR DESCRIPTION
**Pull request by fernandohds564**
Add properties that allow configuring the frequency generator to perform automated and (internally or externally) triggered frequency sweeps.

### CODE REVIEW

File: siriuspy/siriuspy/devices/rf.py [hunk 1]
```diff
@@ -3,24 +3,92 @@
...
 from .device import DeviceNC as _DeviceNC
 from .device import Devices as _Devices
 
 
 class RFGen(_DeviceNC):
-    """."""
+    """Encapsulate the basic features of the RF Generator used in Sirius.
+
+    The Generator is the R&S SMA100A Signal Generator.
+    For more informations of the funcionalities of this equipment, please
```
Comment (anacso17) [inline siriuspy/siriuspy/devices/rf.py:16]: typo ` functionalities `

File: siriuspy/siriuspy/devices/rf.py [hunk 2]
```diff
@@ -3,24 +3,92 @@
...
 
     RF_DELTA_MIN = 0.1  # [Hz]
     RF_DELTA_MAX = 15000.0  # [Hz]
     RF_DELTA_RMP = 200  # [Hz]
 
+    PHASE_CONTINUOUS = _get_namedtuple('PhaseContinuous', ('OFF', 'ON'))
+    FREQ_OPMODE = _get_namedtuple(
+        'FreqOpMode', (
+            'CW',  # Continuous wave (regular mode of operation.)
+            'SWE',  # Seep Mode turned on
```
Comment (anacso17) [inline siriuspy/siriuspy/devices/rf.py:34]: S**w**eep mode?

File: siriuspy/siriuspy/devices/rf.py [hunk 3]
```diff
@@ -3,24 +3,92 @@
...
+            'LIST')  # I don't know yet.
+        )
+    SWEEP_MODE = _get_namedtuple(
+        'SweepMode', (
+            'AUTO',  # each trigger exectutes waveform
```
Comment (anacso17) [inline siriuspy/siriuspy/devices/rf.py:39]: executes?

File: siriuspy/siriuspy/devices/rf.py [hunk 4]
```diff
@@ -55,6 +128,418 @@ def frequency(self, value):
...
+        self.freq_sweep_shape = self.SWEEP_SHAPE.TRI
+        return self._wait('FreqFreqShp-Sts', self.SWEEP_SHAPE.TRI)
+
+    @property
+    def freq_sweep_retrace(self):
+        """Define whether to return to start freq. after sweep finish.
+
+        Takes the following values:
+            0 - 'OFF': return to starting frequency.
+            1 - 'ON': stay at stop frequency frequency.
```
Comment (anacso17) [inline siriuspy/siriuspy/devices/rf.py:378]: It seems the text here sould be like in row 393: `1 - 'ON': stay at last frequency of the sweep.`

**Review by anacso17 — APPROVED** [review]

File: siriuspy/siriuspy/devices/rf.py (hunk 1 above)
Comment (fernandohds564) [reply siriuspy/siriuspy/devices/rf.py:16]: done!

File: siriuspy/siriuspy/devices/rf.py (hunk 2 above)
Comment (fernandohds564) [reply siriuspy/siriuspy/devices/rf.py:34]: done!

File: siriuspy/siriuspy/devices/rf.py (hunk 3 above)
Comment (fernandohds564) [reply siriuspy/siriuspy/devices/rf.py:39]: done!

File: siriuspy/siriuspy/devices/rf.py (hunk 4 above)
Comment (fernandohds564) [reply siriuspy/siriuspy/devices/rf.py:378]: done!

**Review by xresende — APPROVED** [review]

**Pull request merged**